packages/snowpack/src/util.ts

```diff
@@ -177,22 +177,44 @@ export async function openInBrowser(
     : /brave/i.test(browser)
     ? appNames[process.platform]['brave']
     : browser;
-  if (process.platform === 'darwin' && /chrome|default/i.test(browser)) {
+  const isMac = process.platform === 'darwin';
+  const isOpeningInChrome = /chrome|default/i.test(browser);
+  if (isMac && isOpeningInChrome) {
     // If we're on macOS, and we haven't requested a specific browser,
     // we can try opening Chrome with AppleScript. This lets us reuse an
     // existing tab when possible instead of creating a new one.
     try {
+      // see if Chrome process is open; fail if not
       await execa.command('ps cax | grep "Google Chrome"', {
         shell: true,
       });
-      await execa('osascript ../assets/openChrome.applescript "' + encodeURI(url) + '"', {
-        cwd: __dirname,
-        stdio: 'ignore',
-        shell: true,
-      });
+      // use open Chrome tab if exists; create new Chrome tab if not
+      const openChrome = execa(
+        'osascript ../assets/openChrome.applescript "' + encodeURI(url) + '"',
+        {
+          cwd: __dirname,
+          stdio: 'ignore',
+          shell: true,
+        },
+      );
+      // if Chrome doesn’t respond within 3s, fall back to opening new tab in default browser
+      let isChromeStalled = setTimeout(() => {
+        openChrome.cancel();
+        console.warn(`Chrome not responding to Snowpack after 3s. Opening dev server in new tab.`);
```

Comment:
question: if chrome is not responding, wouldn't this fail on open as well?

Reply:
There’s 3 scenarios here, I guess:
1. Chrome is open but not responding, and not the default browser: this is an improvement
2. Chrome is open and default but something in the AppleScript failed (e.g. Chrome’s API changed): this is an improvement
3. Chrome is open and not responding: you’re right this would hang anyway, but at least now it logs “Chrome isn’t responding” so users would at least know to try and quit it.

Reply:
awesome, makes sense to me :+1: